src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -535,18 +541,172 @@ export const maskUidOnLastField = (
+  const maskedAnswerTemplate = field.answerTemplate.map((answer) => {
+    return answer.length >= 4 // defensive, in case UID length is less than 4
+      ? '*'.repeat(answer.length - 4) + answer.substr(-4)
+      : answer
+  })
+  return {
+    question: field.question,
+    answerTemplate: maskedAnswerTemplate,
+  }
+}
+
+/**
+ * Function to extract information for email json field from response
+ * Json field is used for data collation tool
+ */
+const getJsonFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailJsonField | undefined => {
+  const { answer, fieldType } = response
+  // Headers are excluded from JSON data
+  if (fieldType !== BasicField.Section) {
+    const jsonData = {
+      question: getJsonPrefixedQuestion(response),
+      answer,
+    }
+    return jsonData
+  }
+  return undefined
+}
+
+/**
+ * Function to extract information for email form field from response
+ * Form field is used to send responses to admin
+ */
+const getFormFormattedResponse = (
+  response: ResponseFormattedForEmail,
+  hashedFields: Set<string>,
+): EmailFormField => {
+  const { answer, fieldType } = response
+  const answerSplitByNewLine = answer.split('\n')
+  const formData = {
+    question: getFormDataPrefixedQuestion(response, hashedFields),
+    answerTemplate: answerSplitByNewLine,
+    answer,
+    fieldType,
+  }
+  return formData
+}
+
+/**
+ * Function to extract information for email autoreply field from response
+ * Autoreply field is used to send confirmation emails
+ */
+const getAutoReplyFormattedResponse = (
+  response: ResponseFormattedForEmail,
+): EmailAutoReplyField | undefined => {
+  const { question, answer, isVisible } = response
+  const answerSplitByNewLine = answer.split('\n')
+  // Auto reply email will contain only visible fields
+  if (isVisible) {
+    const autoReplyData = {
+      question, // No prefixes for autoreply
+      answerTemplate: answerSplitByNewLine,
+    }
+    return autoReplyData
+  }
+  return undefined
+}
+
+export class EmailDataObj {
+  parsedResponses: ProcessedFieldResponse[]
+  hashedFields: Set<string>
+  authType: AuthType
+
+  constructor(
+    parsedResponses: ProcessedFieldResponse[],
+    hashedFields: Set<string>,
+    authType: AuthType,
+  ) {
+    this.parsedResponses = parsedResponses
+    this.hashedFields = hashedFields
+    this.authType = authType
+  }
+
+  /**
+   * Getter function to return jsonData which is used for data collation tool
+   */
+  get jsonData(): EmailJsonField[] {
+    return this.parsedResponses.flatMap((response) =>
+      compact(
+        createFormattedDataForOneField(
+          response,
+          this.hashedFields,
+          getJsonFormattedResponse,
+        ),
+      ),
+    )
+  }
+
+  /**
+   * Getter function to return autoReplyData for confirmation emails to respondent
+   * If AuthType is CP, return a masked version
+   */
+  get autoReplyData(): EmailAutoReplyField[] {
```

Comment:
might this be a good time to change the `autoReply` naming convention as well? we all understand what "autoreply" refers to because we're used to it, but the name of the feature is really "email confirmation", and we constantly have to explain the word "autoreply" to new team members

Reply:
^see comments above